### PR TITLE
issue/2558 Added contentObject ready wait queue and inview ready lock

### PR DESCRIPTION
**Pull request by oliverfoster**
#2558 
* Added a wait queue around content object ready functions so that extensions can hold the loading screen until they are rendered correctly
* Lock inview until content object is ready so that inview doesn't trigger on initial text blocks until ready

### CODE REVIEW

File: src/core/js/views/menuView.js [hunk 1]
```diff
@@ -28,26 +29,33 @@ define([
...
+
+            var performIsReady = function() {
+                $('.loading').hide();
+                $(window).scrollTop(0);
+                Adapt.trigger('menuView:ready', this);
+                $.inview.unlock('menuView');
+                var styleOptions = { opacity: 1 };
+                if (this.disableAnimation) {
+                    this.$el.css(styleOptions);
+                    $.inview.unlock('menuView');
```
Comment (cahirodoherty-learningpool) [inline src/core/js/views/menuView.js]: You are calling this twice

Comment (oliverfoster) [reply src/core/js/views/menuView.js]: I'd misplaced it originally. Good catch.

File: src/core/js/views/pageView.js [hunk 2]
```diff
@@ -22,32 +22,39 @@ define([
...
+
+            var performIsReady = function() {
+                $('.loading').hide();
+                $(window).scrollTop(0);
+                Adapt.trigger('pageView:ready', this);
+                $.inview.unlock('pageView');
+                var styleOptions = { opacity: 1 };
+                if (this.disableAnimation) {
+                    this.$el.css(styleOptions);
+                    $.inview();
```
Comment (moloko) [inline src/core/js/views/pageView.js]:
```suggestion
                    $.inview();
                    $(window).scroll();
                    return;
```
then you can lose the `else` block

File: src/core/js/views/menuView.js [hunk 3]
```diff
@@ -28,26 +29,32 @@ define([
...
+
+            var performIsReady = function() {
+                $('.loading').hide();
+                $(window).scrollTop(0);
+                Adapt.trigger('menuView:ready', this);
+                $.inview.unlock('menuView');
+                var styleOptions = { opacity: 1 };
+                if (this.disableAnimation) {
+                    this.$el.css(styleOptions);
+                    $.inview();
```
Comment (moloko) [inline src/core/js/views/menuView.js]:
```suggestion
                    $.inview();
                    $(window).scroll();
                    return;
```
and drop the `else`?

**Pull request merged**